### DISCUSSION TITLE
i3bar is displayed twice after regolith packages update

### DISCUSSION THREAD

**Discussion by mwmdev** [regolith-linux/regolith-desktop#880] [Q&A]
I just updated these packages :

> 2023-08-04 17:46:53 install i3-swap-focus:amd64 <none> 0.4.3-1regolith
> 2023-08-04 17:47:13 install regolith-look-default-loader:amd64 <none> 0.7.5-1regolith
> 2023-08-04 17:47:14 install regolith-i3-control-center-regolith:amd64 <none> 3.8.1-2-1regolith-jammy
> 2023-08-04 17:47:14 install regolith-i3-gaps-partial:amd64 <none> 3.8.1-2-1regolith-jammy
> 2023-08-04 17:47:15 install regolith-i3-i3xrocks:amd64 <none> 3.8.1-2-1regolith-jammy
> 2023-08-04 17:47:15 install regolith-i3-ilia:amd64 <none> 3.8.1-2-1regolith-jammy
> 2023-08-04 17:47:15 install regolith-i3-rofication-ilia:amd64 <none> 3.8.1-2-1regolith-jammy
> 2023-08-04 17:47:15 install regolith-i3-swap-focus:amd64 <none> 3.8.1-2-1regolith-jammy

and now my i3bar is displayed twice one under the other : 

![image](https://github.com/regolith-linux/regolith-desktop/assets/31964517/15c743d2-0320-4125-b190-c5e15787bf3c)

I see two `i3bar` processes running :

```
$ps aux | grep i3bar
mike      141331  0.1  0.0 273264 13896 ?        S    21:28   0:01 i3bar --bar_id=bar-0 --socket=/run/user/1000/i3/ipc-socket.2637
mike      141333  0.0  0.0 271940 11876 ?        S    21:28   0:01 i3bar --bar_id=bar-1 --socket=/run/user/1000/i3/ipc-socket.2637
```

Can't figure out where the duplicate is called from.

**Comment by kgilmer** [comment]
Hi @mwmdev , which version of Regolith are you running?  Are you getting packages from `unstable` by chance? (`/etc/apt/sources.list.d/regolith.list` might have "release" "testing" or "unstable" in the URL path)

To answer your question, the package that provides the bar stanza is `regolith-i3-i3xrocks`.  You could determine where the dupe is by first finding all files that i3 is loading:

```shell
$ i3 --moreversion
```

This will show each file and directory.  then you can grep through what those are looking for `bar {` in the files.  Ex:

```shell
$ find /usr/share/regolith/i3/config.d/ | xargs grep -HsF "bar {"
/usr/share/regolith/i3/config.d/70_bar:bar {
```

Unfortunately this doesn't tell you which deb package is supplying the file.  To do that you can then ask `dpkg`:

```shell
$ dpkg -S 70_bar
regolith-i3-i3xrocks: /usr/share/regolith/i3/config.d/70_bar
regolith-sway-i3status-rs: /usr/share/regolith/sway/config.d/70_bar_i3status-rs
```

In the case that you don't find a package corresponding to a file, it means that the file was probably supplied by you or was deprecated from a previous release but not removed.  HTH

**Comment by mwmdev** [comment]
Hi, thanks for the thorough reply. It was a simple mistake, I had custom configs in .config/ and forgot to delete the one in /usr/share, so the bar was loaded twice.
